### PR TITLE
Waveforms Notebook Internal Review

### PR DESCRIPTION
**Pull request by katrinaager**
This is the classify_waveforms notebook that needs editing suggestions.

### CODE REVIEW

**Comment by katrinaager** [comment]
@jeromelecoq please review the newest commit I just pushed, it is the new draft based on carter's feedback

File: docs/first-order/classify_waveforms.ipynb [hunk 1]
```diff
@@ -0,0 +1,2327 @@
+{
```
Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>bolden this warning</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    column_names = electrodes.colnames
</pre><p>unused variable</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # function that returns a list of channels ids that are valid within electrodes table
</pre><p>can remove 'function that'. You want comments to be as brief as possible, but not more brief.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>In the first sentence, don't say "generate a list of data" from the units table. Be precise about what the output is. In this case, since it can be used for many kinds of data, you could be more explicit about that. This function actually outputs a list of <em>indices. </em>You mention this further down so its good, but maybe rephrase to first part to be clear that this could be used to access unit attributes rather than just "data".</p><p> </p><p>Typo in "for-loop that loops through the..."</p><p> </p><p>Typo in "superceeding"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>dont need to print all waveform durations</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>you can use the selected indices here to vastly shorten this code, make it more readable, and make it more efficient.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    electrodes = nwb.electrodes.to_dataframe()
</pre><p>this should be in a different cell. Don't you already cast it to a dataframe earlier?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>Write the name of the optotagging notebook. Maybe give a one or two sentence explanation of what this table is.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>same thing as bi code block above. Can use the selected unit indices here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>If you want users to do this with FS and RS, write another line <code>units_spike_times = unit_spike_times_rs</code> and comment it out. then add another comment telling the user they can swap these lines. also mention it in the markdown. you can remove the  <code>#units_spike_times = units['spike_times']</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>mention it in the markdown that they can change censor_period, stim_duration, time_resolution, and window times. Also would be best to put those in their own cell above this one. Probably should also add a new section heading to explain this stuff? in the heading you can also briefly explain what this code is doing, and mention the purpose of the function 'show_counts' and what its output means</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (jeromelecoq) [inline docs/first-order/classify_waveforms.ipynb:1]: <blockquote>classify the difference between fast-spiking and regular-spiking neurons</blockquote><blockquote><br></blockquote><p>Try to re-read and remove unnecessary duplicated ideas."Classify the difference" does not mean much. You are either "classifying the two types of neurons" or Highlight their difference.</p><p>Rephrase.</p><p> </p><p>Also" will utilize open source data "</p><p>Try to make your verb focus on the most important thing. In this notebook, you are mostly classifying  FS and RS using Allen Institute as example data. The primary purpose is not to utilize open source data. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (jeromelecoq) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>Do we need to print all of them? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (jeromelecoq) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>I suggest to re-shuffle a bit the paragraphs. In this paragraph you should be introducing the next bit of code, which is about sub-selecting cells. The thresholding is happening later on when you have the distribution. This is when the explanation would be useful. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (jeromelecoq) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>I suggest to add a citation about this threshold. Corbett can probably recommend a good source. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (jeromelecoq) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>I think it would make sense to compare RS and FS cells for optotagging.</p><p>The hypotheses is that PV+ cells (responsive to opt-tagging ) should also be FS cells. You should state it and confirm.</p><p> </p><p>I would make this comparison much more apparent. Perhaps showing the two plots side by side. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (rcpeene) [reply docs/first-order/classify_waveforms.ipynb:1]: <p>ya dont print the whole thing</p>

File: docs/first-order/classify_waveforms.ipynb [hunk 2]
```diff
@@ -0,0 +1,2417 @@
...
```
Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    data = electrodes.to_dataframe()
</pre><p>variable shouldnt be called "data" u can just call it electrodes_df</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>great description. Capitalize <code>Units</code> and <code>Electrodes</code>  table names</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

File: docs/first-order/classify_waveforms.ipynb [hunk 3]
```diff
@@ -0,0 +1,2107 @@
...
```
Comment (rcpeene) [inline docs/first-order/classify_waveforms.ipynb:1]: <p>typo</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AllenInstitute/openscope_databook/pull/281/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>
